File: frontend/src/types/conversation.ts
export interface ConversationResponse {
  id: string
  title: string
  mode: string
  agent_id: string | null
  status: string
  model_name: string | null
  total_tokens: number
  starred: boolean
  created_at: string
  updated_at: string | null
}

export interface MessageResponse {
  id: string
  role: string
  content: string | null
  message_type: string
  metadata: Record<string, unknown> | null
  created_at: string
}

export interface ConversationDetail extends ConversationResponse {
  messages: MessageResponse[]
}

export interface ConversationCreate {
  title?: string
  mode: "react" | "dag"
  agent_id?: string
  model_name?: string
}

export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  size: number
  pages: number
}

File: src/fim_agent/web/schemas/conversation.py
"""Conversation and message request/response schemas."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class ConversationCreate(BaseModel):
    title: str = ""
    mode: Literal["react", "dag"] = "react"
    agent_id: str | None = None
    model_name: str | None = None


class ConversationUpdate(BaseModel):
    title: str | None = None
    status: Literal["active", "archived"] | None = None
    starred: bool | None = None


class BatchDeleteRequest(BaseModel):
    ids: list[str] = Field(..., max_length=50)


class ConversationResponse(BaseModel):
    id: str
    title: str
    mode: str
    agent_id: str | None
    status: str
    starred: bool
    model_name: str | None
    total_tokens: int
    created_at: str
    updated_at: str | None


class MessageResponse(BaseModel):
    id: str
    role: str
    content: str | None
    message_type: str
    metadata: dict | None
    created_at: str


class ConversationDetail(ConversationResponse):
    messages: list[MessageResponse] = []
